attributes-info/src/builtin_attributes.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs::File;
+
+use serde::{Serialize, Serializer};
+use serde::ser::SerializeSeq;
+
+use rustc_feature::{AttributeDuplicates, AttributeGate, AttributeType, BUILTIN_ATTRIBUTES};
+
+use crate::formatter::MyFormatter;
+
+#[derive(Serialize)]
+struct BuiltinAttributeInfo {
+    name: &'static str,
+    #[serde(rename = "type")]
+    type_: MyAttributeType,
+    template: MyAttributeTemplate,
+    duplicates: &'static str,
+    gated: bool,
+}
+
+#[derive(Serialize)]
+pub enum MyAttributeType {
+    Normal,
+    CrateLevel,
+}
+
+#[derive(Serialize)]
+pub struct MyAttributeTemplate {
+    /// If `true`, the attribute is allowed to be a bare word like `#[test]`.
+    pub word: bool,
+    /// If `Some`, the attribute is allowed to take a list of items like `#[allow(..)]`.
+    pub list: Option<&'static str>,
+    /// If `Some`, the attribute is allowed to be a name/value pair where the
+    /// value is a string, like `#[must_use = "reason"]`.
+    #[serde(rename = "nameValueStr")]
+    pub name_value_str: Option<&'static str>,
+}
+
+pub(crate) fn generate_builtin_attributes_json(builtin_attributes_json_path: &str) {
+    let f = File::create(builtin_attributes_json_path).unwrap();
+    let mut serializer = serde_json::Serializer::with_formatter(f, MyFormatter::new());
+
+    let mut seq = serializer.serialize_seq(None).unwrap();
+    for attr in BUILTIN_ATTRIBUTES {
```

Comment:
What do you think about sorting attributes in our json in alphabetical order?

Reply:
Hmm, sounds reasonable